modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -0,0 +1,510 @@
+      else if (!_ring_block_ids[i].empty() &&
+               _ring_block_ids[i].size() != (_ring_radii[i].size() + (_ring_intervals[i][0] > 1)))
+        paramError("ring_block_ids",
+                   "The parameter must be consistent with ring_radii if provided.");
+      if (!_has_rings[i] && !_ring_block_names[i].empty())
+        paramError("ring_block_names",
+                   "The parameter must be consistent with ring_radii if provided.");
+      else if (!_ring_block_names[i].empty() &&
+               _ring_block_names[i].size() != (_ring_radii[i].size() + (_ring_intervals[i][0] > 1)))
+        paramError("ring_block_names",
+                   "The parameter must be consistent with ring_radii if provided.");
+    }
+  if (!_background_block_ids.empty())
+  {
+    if (_has_rings[0] && _has_rings[1] && _has_rings[2] && _background_block_ids.size() != 1)
+      paramError("background_block_ids",
+                 "If provided, the size of this parameter must be one if all sections have rings.");
+    else if (!_has_rings[0] && !_has_rings[1] && !_has_rings[2] && _background_intervals == 1 &&
+             _background_block_ids.size() != 1)
+      paramError("background_block_ids",
+                 "If provided, the size of this parameter must be one if no sections have rings "
+                 "and background_intervals is one.");
+    else if ((!_has_rings[0] || !_has_rings[1] || !_has_rings[2]) &&
+             _background_block_ids.size() != 2)
+      paramError(
+          "background_block_ids",
+          "If provided, the size of this parameter must be two if ring-free section exists.");
+  }
+  if (!_background_block_names.empty())
+  {
+    if (_has_rings[0] && _has_rings[1] && _has_rings[2] && _background_block_names.size() != 1)
+      paramError("background_block_names",
+                 "If provided, the size of this parameter must be one if all sections have rings.");
+    else if (!_has_rings[0] && !_has_rings[1] && !_has_rings[2] && _background_intervals == 1 &&
+             _background_block_names.size() != 1)
+      paramError("background_block_names",
+                 "If provided, the size of this parameter must be one if no sections have rings "
+                 "and background_intervals is one.");
+    else if ((!_has_rings[0] || !_has_rings[1] || !_has_rings[2]) &&
+             _background_block_names.size() != 2)
+      paramError(
+          "background_block_names",
+          "If provided, the size of this parameter must be two if ring-free section exists.");
+  }
+  if (_pin_id_name.empty())
+  {
+    if (!_pin_id_values.empty())
+      paramError("pin_id_values",
+                 "This parameter cannot be used when pin_id_name is not provided.");
+  }
+  else if (_pin_id_values.size() != 3)
+    paramError(
+        "pin_id_values",
+        "If pin_id_name is provided, this parameter must be provided with a length of three.");
+
+  // Improve before submitting
```

Comment:
^^
yeah we need to check the offset + pin max radius is still inside the hexagon.
and offset - pin_max_radius dont overlap at the center for the 3 pins right?

Reply:
Weird, I remember fixing this. I may have missed some commit.

Reply:
still need to remove that comment

Reply:
We added some comment say that additional check is done elsewhere.